src/lib.rs

```diff
@@ -0,0 +1,8 @@
+use blake3::hash;
+
+#[no_mangle]
+extern "C" fn blake3(mem: &mut [u8; 32], len: usize, input: *const u8) {
+  unsafe {
+    *mem = *hash(std::slice::from_raw_parts(input, len)).as_bytes();
```

Comment:
Could we convert the ByteArray to a Rust slice more directly, i.e. passing in`oa->m_data` from Lean and having the Rust function take in a `&[u8]`?